.devops/nix/python-scripts.nix

```diff
@@ -0,0 +1,43 @@
+{
+  lib,
+  stdenv,
+  buildPythonPackage,
+  poetry-core,
+  breakpointHook,
+  mkShell,
+  python3Packages,
+  gguf-py,
+}@inputs:
+
+let
+  llama-python-deps = with python3Packages; [
+    numpy
+    sentencepiece
+    transformers
+    protobuf
+    torchWithoutCuda
+    gguf-py
+    tqdm
+  ];
+in
+
+buildPythonPackage ({
+  pname = "llama-scripts";
+  src = ../../.;
+  version = "0.0.0";
+  pyproject = true;
+  nativeBuildInputs = [ poetry-core ];
+  projectDir = ../../.;
+  propagatedBuildInputs = llama-python-deps;
```

Comment:
What's projectDir for?

Reply:
`src = ../../.` will trigger many unnecessary rebuilds, e.g. when editing nix or yaml files

Reply:
It's customary to sort the attributes in the phase order and to put `pname version pyproject=true src` at the top.

Reply:
Nixpkgs might be preparing to switch from build-time to eval-time propagation of python dependencies, and and within that effort it's recommended to use `dependencies = [ ... ]` instead of `propagatedBuildInputs = [ ... ]` (`buildPythonPackage` currently just merges `dependencies` back into `propagatedBuildInputs` so it's an alias, but this might switch to a "smarter" logic in future)

Reply:
> What's projectDir for?

Totally redundant, removed

> src = ../../. will trigger many unnecessary rebuilds, e.g. when editing nix or yaml files

Good point, I've added filtering for non-source files

> recommended to use dependencies = [ ... ] instead of propagatedBuildInputs = [ ... ]

Nice to know! Done